ipaserver/install/server/replicainstall.py

```diff
@@ -448,9 +448,9 @@ def promote_sssd(host_name):
     if ipa_domain is None:
         raise RuntimeError("Couldn't find IPA domain in sssd.conf")
     else:
-        domain.set_option('ipa_server', host_name)
-        domain.set_option('ipa_server_mode', True)
-        sssdconfig.save_domain(domain)
+        ipa_domain.set_option('ipa_server', host_name)
+        ipa_domain.set_option('ipa_server_mode', True)
+        sssdconfig.save_domain(ipa_domain)
```

Comment:
IMHO this is a better fix:
```diff
diff --git a/ipaserver/install/server/replicainstall.py b/ipaserver/install/server/replicainstall.py
index 9d251b5e9..e80584cf0 100644
--- a/ipaserver/install/server/replicainstall.py
+++ b/ipaserver/install/server/replicainstall.py
@@ -437,2 +436,0 @@ def promote_sssd(host_name):
-    ipa_domain = None
-
@@ -444 +442 @@ def promote_sssd(host_name):
-                ipa_domain = domain
+                break
@@ -447,3 +444,0 @@ def promote_sssd(host_name):
-
-    if ipa_domain is None:
-        raise RuntimeError("Couldn't find IPA domain in sssd.conf")
@@ -451,4 +446 @@ def promote_sssd(host_name):
-        domain.set_option('ipa_server', host_name)
-        domain.set_option('ipa_server_mode', True)
-        sssdconfig.save_domain(domain)
-        sssdconfig.write()
+        raise RuntimeError("Couldn't find IPA domain in sssd.conf")
@@ -456,5 +448,10 @@ def promote_sssd(host_name):
-        sssd = services.service('sssd', api)
-        try:
-            sssd.restart()
-        except CalledProcessError:
-            logger.warning("SSSD service restart was unsuccessful.")
+    domain.set_option('ipa_server', host_name)
+    domain.set_option('ipa_server_mode', True)
+    sssdconfig.save_domain(domain)
+    sssdconfig.write()
+
+    sssd = services.service('sssd', api)
+    try:
+        sssd.restart()
+    except CalledProcessError:
+        logger.warning("SSSD service restart was unsuccessful.")
```
It removes both redundant variable and an `else` branch but if you don't like it, I can ACK this as well.

Reply:
I liked your proposal, so squashing commits and re-basing.